### PR TITLE
audio: host-zephyr: Allocate DMA block configuration on heap

### PR DESCRIPTION
**Pull request by LaurentiuM1234**
Currently, the DMA block configuration (struct dma_block_config) is placed on the stack during host_common_params(). Later on, host_copy_one_shot() tries to re-configure the DMAC driver using the same block configuration that's located in a stack frame that's no longer valid. As such, allocate a DMA block configuration on the heap during host_common_params() which can be used later on (e.g: during host_copy_one_shot()) to reconfigure the DMAC.

Cherry-picked from https://github.com/thesofproject/sof/pull/8859

### CODE REVIEW

**Review by dbaluta — APPROVED** [review]

**Comment by dbaluta** [comment]
@marc-hb @lgirdwood I think I'm out of the loop here, what is the status of fixing zephyr builds? Are we waiting for some change to be merged into zephyr right?

**Comment by marc-hb** [comment]
> what is the status of fixing zephyr builds? Are we waiting for some change to be merged into zephyr right?

This is tracked in #8818

**Review by iuliana-prodan — APPROVED** [review]

**Review by kv2019i — APPROVED** [review]

File: src/audio/host-zephyr.c [hunk 1]
```diff
@@ -912,23 +920,24 @@ int host_common_params(struct host_data *hd, struct comp_dev *dev,
...
-		dma_block_cfg.dest_address = buffer_addr;
+		dma_block_cfg->dest_address = buffer_addr;
 		break;
 	}
 
 	err = dma_config(hd->chan->dma->z_dev, hd->chan->index, dma_cfg);
 	if (err < 0) {
 		comp_err(dev, "host_params(): dma_config() failed");
 		dma_release_channel(hd->dma->z_dev, hd->chan->index);
+		rfree(dma_block_cfg);
```
Comment (lyakh) [inline src/audio/host-zephyr.c:940]: do we also want to reset `dma_cfg->head_block` to `NULL` before freeing? Might be worth making an error handling label at the bottom of the function

Comment (LaurentiuM1234) [reply src/audio/host-zephyr.c]: fixed, should be ok now

**Review by lgirdwood — APPROVED** [review]

**Review by lyakh — APPROVED** [review]

**Review by abonislawski — APPROVED** [review]

**Pull request merged**